designer/server/src/test/scala/pl/touk/nussknacker/ui/api/AppResourcesSpec.scala

```diff
@@ -1,198 +1,197 @@
-    createDeployedCanceledProcess(ProcessName("id1"))
-    createDeployedProcess(ProcessName("id2"))
-    val resources = prepareBasicAppResources(
-      statuses = Map(
+    createDeployedCanceledProcess(ProcessName("id1"), category = Category1)
+    createDeployedProcess(ProcessName("id2"), category = Category1)
+
+    MockableDeploymentManager.configure(
+      Map(
         ProcessName("id2") -> ProblemStateStatus.shouldBeRunning(VersionId(1L), "user"),
       )
     )
 
-    val result = Get("/app/healthCheck/process/deployment") ~> withPermissions(resources, testPermissionRead)
-
-    result ~> check {
-      status shouldBe StatusCodes.InternalServerError
-      val expectedResponse = HealthCheckProcessResponse(ERROR, Some("Scenarios with status PROBLEM"), Some(Set("id2")))
-      entityAs[HealthCheckProcessResponse] shouldBe expectedResponse
-    }
+    given()
+      .auth().basic("reader", "reader")
+      .when()
+      .get(s"$nuDesignerHttpAddress/api/app/healthCheck/process/deployment")
+      .Then()
+      .statusCode(500)
+      .body(equalsJson(
+        s"""{
+           |  "status": "ERROR",
+           |  "message": "Scenarios with status PROBLEM",
+           |  "processes": [ "id2" ]
+           |}""".stripMargin
+      ))
   }
 
   test("it should return healthcheck ok if statuses are ok") {
-    createDeployedProcess(ProcessName("id1"))
-    createDeployedProcess(ProcessName("id2"))
-    val resources = prepareBasicAppResources(
-      statuses = Map(
+    createDeployedProcess(ProcessName("id1"), category = Category1)
+    createDeployedProcess(ProcessName("id2"), category = Category1)
+
+    MockableDeploymentManager.configure(
+      Map(
         ProcessName("id1") -> SimpleStateStatus.Running,
         ProcessName("id2") -> SimpleStateStatus.Running,
       )
     )
 
-    val result = Get("/app/healthCheck/process/deployment") ~> withPermissions(resources, testPermissionRead)
-
-    result ~> check {
-      status shouldBe StatusCodes.OK
-    }
+    given()
+      .auth().basic("reader", "reader")
+      .when()
+      .get(s"$nuDesignerHttpAddress/api/app/healthCheck/process/deployment")
+      .Then()
+      .statusCode(200)
+      .body(equalsJson(
+        s"""{
+           |  "status": "OK",
+           |  "message": null,
+           |  "processes": null
+           |}""".stripMargin
+      ))
   }
 
   test("it should not report deployment in progress as fail") {
     createDeployedProcess(ProcessName("id1"))
-    val resources = prepareBasicAppResources(
-      statuses = Map(
-        ProcessName("id1") -> SimpleStateStatus.Running,
+    createDeployedCanceledProcess(ProcessName("id1"))
+
+    MockableDeploymentManager.configure(
+      Map(
+        ProcessName("id1") -> SimpleStateStatus.Running
       )
     )
 
-    val result = Get("/app/healthCheck/process/deployment") ~> withPermissions(resources, testPermissionRead)
-
-    result ~> check {
-      status shouldBe StatusCodes.OK
-    }
+    given()
```

Comment:
I think you cane move whole code to a function with response code and responseBody?

Reply:
what will be the purpose of the change you propose?

Reply:
I think that @lciolecki would like to optimize the amount of code, and BDD is standing in opposition to drastic repetition reductions. We should pick one of two: DRY as much as you can vs BDD with a little dose of repetition. I like the second approach, what about you @lciolecki ?

Reply:
IMO in tests the approach DRY is not always right. One of the most important things is the readability of tests. Currently, thanks to rest-assured DSL, we have a nice given-when-then approach, actually forced by the lib. 

So, I agree with @arkadius and I'd stay with the current approach.